homeassistant/components/fritzbox_callmonitor/config_flow.py

```diff
@@ -0,0 +1,255 @@
+    DOMAIN,
+    FRITZ_ACTION_GET_INFO,
+    FRITZ_ATTR_NAME,
+    FRITZ_ATTR_SERIAL_NUMBER,
+    FRITZ_SERVICE_DEVICE_INFO,
+    SERIAL_NUMBER,
+)
+
+DATA_SCHEMA_USER = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+        vol.Required(CONF_USERNAME, default=DEFAULT_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+RESULT_INVALID_AUTH = "invalid_auth"
+RESULT_INSUFFICIENT_PERMISSIONS = "insufficient_permissions"
+RESULT_MALFORMED_PREFIXES = "malformed_prefixes"
+RESULT_NO_DEVIES_FOUND = "no_devices_found"
+RESULT_SUCCESS = "success"
+
+
+class FritzBoxCallMonitorConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a fritzbox_callmonitor config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize flow."""
+        self._host = None
+        self._port = None
+        self._username = None
+        self._password = None
+        self._phonebook_name = None
+        self._phonebook_id = None
+        self._phonebook_ids = None
+        self._phonebook = None
+        self._prefixes = None
+        self._serial_number = None
+
+    def _get_config_entry(self):
+        """Create and return an config entry."""
+        return self.async_create_entry(
+            title=self._phonebook_name,
+            data={
+                CONF_HOST: self._host,
+                CONF_PORT: self._port,
+                CONF_USERNAME: self._username,
+                CONF_PASSWORD: self._password,
+                CONF_PHONEBOOK: self._phonebook_id,
+                CONF_PREFIXES: self._prefixes,
+                SERIAL_NUMBER: self._serial_number,
+            },
+        )
+
+    def _try_connect(self):
+        """Try to connect and check auth."""
+        self._phonebook = FritzBoxPhonebook(
+            host=self._host,
+            username=self._username,
+            password=self._password,
+            phonebook_id=self._phonebook_id,
+            prefixes=self._prefixes,
+        )
+
+        try:
+            self._phonebook.init_phonebook()
+            self._phonebook_ids = self._phonebook.get_phonebook_ids()
+
+            fritz_connection = FritzConnection(
+                address=self._host, user=self._username, password=self._password
+            )
+            device_info = fritz_connection.call_action(
+                FRITZ_SERVICE_DEVICE_INFO, FRITZ_ACTION_GET_INFO
+            )
+            self._serial_number = device_info[FRITZ_ATTR_SERIAL_NUMBER]
+
+            return RESULT_SUCCESS
+        except RequestsConnectionError:
+            return RESULT_NO_DEVIES_FOUND
+        except FritzSecurityError:
+            return RESULT_INSUFFICIENT_PERMISSIONS
+        except FritzConnectionException:
+            return RESULT_INVALID_AUTH
+
+    def _is_already_configured(self, host, phonebook_id):
```

Comment:
Instead of doing this, set a unique ID: `await self.async_set_unique_id(f"{serial}-{phonebook_id}")`

Reply:
Used in https://github.com/home-assistant/core/pull/40736/commits/00f2ff60a614c1c0bb38150a1d47d6bf79616fc6